From 77d19bb0dd3c2de0e508789eb1224de6a385bdf7 Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Sat, 28 Mar 2020 08:08:28 +0530
Subject: [PATCH 0678/1921] coresight: Correct OcteonTx2 QUIRK_BUFFSIZE_8BX
 applicability

CSETR_QUIRK_BUFFSIZE_8BX is applicable only to OcteonTx2
96xx and 95xx.

Change-Id: If179ab232942592d671e8b3ea0e439680b102a07
Signed-off-by: Linu Cherian <lcherian@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/25774
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/32253
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/hwtracing/coresight/coresight-quirks.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hwtracing/coresight/coresight-quirks.c b/drivers/hwtracing/coresight/coresight-quirks.c
index debef67a592e..2fa5bccfabfb 100644
--- a/drivers/hwtracing/coresight/coresight-quirks.c
+++ b/drivers/hwtracing/coresight/coresight-quirks.c
@@ -17,12 +17,12 @@ u32 coresight_get_etr_quirks(unsigned int id)
 				     MIDR_CPU_VAR_REV(0, 0),
 				     MIDR_CPU_VAR_REV(2, 0)))
 		options |= CSETR_QUIRK_RESET_CTL_REG |
+			CSETR_QUIRK_BUFFSIZE_8BX |
 			CSETR_QUIRK_NO_STOP_FLUSH;
 
 	/* Common across all Chip variants and revisions */
 	if (id == OCTEONTX_CN9XXX_ETR)
 		options |= CSETR_QUIRK_SECURE_BUFF |
-			CSETR_QUIRK_BUFFSIZE_8BX |
 			CSETR_QUIRK_FORCE_64B_DBA_RW;
 
 	return options;
-- 
2.31.1

